pkg/ibc/ibc.go

```diff
@@ -50,15 +67,94 @@ func GetClientsInfo(ibc *relayer.IBCdata, rpcs *map[string]config.RPC) (ClientsI
 
 	ctx := context.Background()
 
-	clientsInfo.ChainAClientExpiration, clientsInfo.ChainAClientInfo, err = relayer.QueryClientExpiration(ctx, chainA, chainB)
+	clientsInfo.ChainAClientExpiration, clientsInfo.ChainAClientInfo, err = relayer.QueryClientExpiration(
+		ctx,
+		chainA,
+		chainB,
+	)
 	if err != nil {
 		return ClientsInfo{}, fmt.Errorf("Error: %w path %v <-> %v", err, cdA, cdB)
 	}
 
-	clientsInfo.ChainBClientExpiration, clientsInfo.ChainBClientInfo, err = relayer.QueryClientExpiration(ctx, chainB, chainA)
+	clientsInfo.ChainBClientExpiration, clientsInfo.ChainBClientInfo, err = relayer.QueryClientExpiration(
+		ctx,
+		chainB,
+		chainA,
+	)
 	if err != nil {
 		return ClientsInfo{}, fmt.Errorf("Error: %w path %v <-> %v", err, cdB, cdA)
 	}
 
 	return clientsInfo, nil
 }
+
+func GetChannelsInfo(ibc *relayer.IBCdata, rpcs *map[string]config.RPC) (ChannelsInfo, error) {
+	ctx := context.Background()
+	channelInfo := ChannelsInfo{}
+
+	if (*rpcs)[ibc.Chain1.ChainName].ChainID == "" || (*rpcs)[ibc.Chain2.ChainName].ChainID == "" {
```

Comment:
there is still
```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x100d172be]

goroutine 107 [running]:
go.uber.org/zap.(*Logger).check(0x0, 0x2, {0x101f5b4b4, 0x1d})
	/Users/tnowak/go/pkg/mod/go.uber.org/zap@v1.24.0/logger.go:298 +0x5e
go.uber.org/zap.(*Logger).Error(0x102a92358?, {0x101f5b4b4?, 0xc001680c80?}, {0xc001698dc0, 0x1, 0x1})
	/Users/tnowak/go/pkg/mod/go.uber.org/zap@v1.24.0/logger.go:235 +0x38
github.com/cosmos/relayer/v2/relayer.UnrelayedSequences({0x102a92358?, 0x103ed52e0}, 0xc001680c80, 0xc0026686e0, 0xc001ca2770)
	/Users/tnowak/go/pkg/mod/github.com/cosmos/relayer/v2@v2.4.1/relayer/naive-strategy.go:25 +0x23c
github.com/archway-network/relayer_exporter/pkg/ibc.GetChannelInfo(0xc0010cc120, 0xc001220290)
	/Users/tnowak/code/archway-network/relayer_exporter/pkg/ibc/ibc.go:148 +0x659
github.com/archway-network/relayer_exporter/pkg/collector.IBCCollector.Collect.func1(0xc0010cc120)
	/Users/tnowak/code/archway-network/relayer_exporter/pkg/collector/collector.go:128 +0x545
created by github.com/archway-network/relayer_exporter/pkg/collector.IBCCollector.Collect in goroutine 92
	/Users/tnowak/code/archway-network/relayer_exporter/pkg/collector/collector.go:102 +0x1b5
exit status 2
```
if a timeout to provided rpc occurs or there is a typo in rpc endpoint.

Reply:
Problem here is that this then needs to be manually verified, that every single connection is valid since `UnrelayedSequences` does not return any errors.

Reply:
There is now check for bad RPC endpoints, this is done via doing a RPC query to them.